Improve documentation about testing modes (#28066)
dotnet/docs: docs/core/tools/dotnet-test.md

  dotnet test --blame
  ```

- Run the tests in the `test1` project providing the `-bl` (binary log) argument down to `msbuild`:

  ```dotnetcli

>   dotnet test ~/projects/test1/test1.csproj -bl

Author: ```suggestion
  dotnet test ~/projects/test1/test1.csproj -bl
  ```
```

>   ```dotnetcli
>   dotnet test ~/projects/test1/test1.csproj -bl
> 

Author: ```suggestion
  ```dotnetcli
  dotnet test ~/projects/test1/test1.csproj -bl
  ```

```

  dotnet test --blame
  ```

> - Run the tests in the `test1` project providing the `-bl` argument down to `msbuild`:
> 
>   ```dotnetcli
>   dotnet test ~/projects/test1/test1.csproj -bl
>   ```

Reviewer: Also consider adding an example of passing an msbuild property

```suggestion
- Run the tests in the `test1` project providing the `-bl` (binary log) argument down to `msbuild`:

  ```dotnetcli
  dotnet test ~/projects/test1/test1.csproj -bl
  ```
```

Author: Isn't this a msbuild property?

Reviewer 2: It's an MSBuild option but not a property.

  dotnet test --blame
  ```

> - Run the tests in the `test1` project providing the `-bl` (binary log) argument down to `msbuild`:

Reviewer 3: ```suggestion
- Run the tests in the `test1` project, providing the `-bl` (binary log) argument to `msbuild`:
```

  ```dotnetcli
  dotnet test ~/projects/test1/test1.csproj -bl  
  ```

> - Run the tests in the `test1` project providing the `-p:DefineConstants="DEV"` argument property down to `msbuild`:

Reviewer 3: ```suggestion
- Run the tests in the `test1` project, setting the MSBuild `DefineConstants` property to `DEV`:
```
